root/components/list/EventList.js

```diff
@@ -72,23 +72,23 @@ const EventList = ({
           false, // to use EntityLink without dates (separate column for that)
         );
       const typeColumn = defineTypeColumn('event_type', order, sortable);
-      const artistsColumn = defineArtistRolesColumn(
+      const artistsColumn = defineArtistRolesColumn<EventT>(
         entity => entity.performers,
         'performers',
         l('Artists'),
       );
-      const timeColumn = defineTextColumn(
+      const timeColumn = defineTextColumn<EventT>(
         entity => entity.time,
         'time',
         l('Time'),
       );
       const rolesOnlyColumn = artist && artistRoles
-        ? defineTextColumn(
+        ? defineTextColumn<EventT>(
           entity => entity.performers.map(performer => (
             performer.entity.id === artist.id
               ? commaOnlyListText(localizeArtistRoles(performer.roles))
-              : null
-          )),
+              : ''
+          )).filter(Boolean).join(', '),
```

Comment:
Instead of using `commaOnlyListText` on each set of roles and then further joining all of those with ", ", it might make more sense if we concatenate all the roles together and use `commaOnlyListText` on the final result. You can do that pretty easily with a `reduce` call, and it also removes the need for the `filter`.
```JS
          entity => commaOnlyListText(entity.performers.reduce((result, performer) => {
            if (performer.entity.id === artist.id) {
              result.push(...localizeArtistRoles(performer.roles));
            }
          }, [])),
```